source/common/router/router_ratelimit.cc

```diff
@@ -174,7 +176,14 @@ bool MaskedRemoteAddressAction::populateDescriptor(RateLimit::DescriptorEntry& d
   // from addressAsString this is a valid address.
   Network::Address::CidrRange cidr_entry =
       *Network::Address::CidrRange::create(remote_address->ip()->addressAsString(), mask_len);
-  descriptor_entry = {"masked_remote_address", cidr_entry.asString()};
+  std::string masked_address = cidr_entry.asString();
+  if (remote_address->ip()->version() == Network::Address::IpVersion::v4) {
+    // For Ipv4, masked_address is returned in a format similar to '192.168.1.0/32', '.' It's a key
+    // word in statsd, This makes it difficult to export measurements in Promethean format in the
+    // RateLimit server.
+    masked_address = absl::StrReplaceAll(masked_address, {{".", "_"}});
```

Comment:
This seems like it might be a breaking change for many people.

Envoy supports many consumers of stats and I feel like we shouldn't tweak naming in a common subsystem to tune to just one of them.

Regarding Prometheus: why not just generate Prom format directly? Why go via statsd?

Reply:
> This seems like it might be a breaking change for many people.

we can either add a runtime feature flag or add an optional API field for this.

> Envoy supports many consumers of stats and I feel like we shouldn't tweak naming in a common subsystem to tune to just one of them.

this problem mainly related to [envoyproxy/ratelimit](https://github.com/envoyproxy/ratelimit), I spend a lot of time in the past trying to bring native promemtheus to it but failed, becase it requires to change a lot of libs used by [envoyproxy/ratelimit](https://github.com/envoyproxy/ratelimit).

Reply:
In general I'd like to see specific needs of individual sinks addressed in the sink, ideally, not in code that generates stat names that would also be used by other sinks.

Reply:
yes, you're right. After rethinking, any ratelimit action that support custom descriptor value will have same problem when user specify `.`.

let me try to fix this in [evnoyproxy/ratelimit](https://github.com/envoyproxy/ratelimit).